### ISSUE TITLE
quote(expr =) should be allowed

### ISSUE THREAD

**Issue by aronatkins** [r-lib/lintr#1889]
The "missing argument" form of `quote(expr =)` causes an `infix_spaces_linter` error. This syntax is suggested by http://adv-r.had.co.nz/Expressions.html#names

If the expression is rewritten as `quote(expr = )`, it causes a `spaces_inside_linter` error.

These errors can be avoided by `nolint` directive comments, but that feels excessive for recommended syntax.

```r
# Code showing both erring syntaxes, and the directive suppressing the issue.
original <- pairlist(path = quote(expr =))
modified <- pairlist(path = quote(expr = ))
avoided <- pairlist(path = quote(expr =)) # nolint: infix_spaces_linter.
```

```bash
# Sample output.
R -s -e 'lintr::lint_dir(".")'
# => examples.R:1:40: style: [infix_spaces_linter] Put spaces around all infix operators.
# => original <- pairlist(path = quote(expr =))
# =>                                        ^
# => examples.R:2:41: style: [spaces_inside_linter] Do not place spaces before parentheses.
# => modified <- pairlist(path = quote(expr = ))
# =>                                         ^
```

**Comment by IndrajeetPatil** [comment]
Note that `{styler}`, which formats code according to tidyverse style guide, adds spaces around infix operators:

``` r
styler::style_text("quote(expr =)")
#> quote(expr = )
```

So the issue is how to not prevent `missing_argument_linter()` from getting triggered with this pattern:

```r
library(lintr)

lint(text = "pairlist(path = quote(expr = ))", linters = all_linters())
#> <text>:1:28: style: [missing_argument_linter] Missing argument in function call.
#> pairlist(path = quote(expr = ))
#>                            ^
```

<sup>Created on 2023-01-05 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

Also, note that you have linked to the first edition of _Advanced R_ where you find this advice. But the second edition recommends using `{rlang}` for missing argument:

``` r
identical(
  pairlist(path = quote(expr = )),
  pairlist(path = rlang::missing_arg())
)
#> [1] TRUE
```

<sup>Created on 2023-01-05 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by aronatkins** [comment]
Thanks for the `rlang` suggestion. I encountered this problem when trying to lint the `packrat` source and it cannot take external dependencies. The implementation of `rlang::missing_arg` relies on a C implementation to return `R_MissingArg` and isn't easily adapted for Packrat:
https://github.com/r-lib/rlang/blob/9e261f86ec8d7aabf8a79705f82881f3447a1ea6/src/internal/exported.c#L696-L698

Funny story: The implementation of `rlang::is_missing` uses `quote(expr = )`.
https://github.com/r-lib/rlang/blob/4b1696c61677b19abdb4a135937e60420ad2b646/R/arg.R#L449

The CRAN release of `lintr` (which does not have `all_linters`):

```r
> lintr::lint(text = "pairlist(path = quote(expr = ))", linters = lintr:::default_linters)
#> <text>:1:29: style: [spaces_inside_linter] Do not place spaces before parentheses.
#> pairlist(path = quote(expr = ))
#>                             ^
```

Feel free to close this as won't-fix if you feel like the use of `quote(expr = )` is rare. It can certainly happen with code that relies only on base R, but most folks are probably fine with an `rlang` dependency.

**Comment by aronatkins** [comment]
One other data point: The `rlang::missing_arg` documentation states that `quote(expr = )` is the canonical way to create a missing argument object.

https://rlang.r-lib.org/reference/missing_arg.html#other-ways-to-reify-the-missing-argument

**Comment by IndrajeetPatil** [comment]
I would say that `{lintr}` should support this pattern. 

I think the easiest way would be to include `quote()` in the vector of exceptions:

https://github.com/r-lib/lintr/blob/6d892a3d3253e36c23e5062792f245d2fd34ab1c/R/missing_argument_linter.R#L34
